gradient_descent.js

```diff
@@ -62,27 +62,76 @@ class AlgorithmFirstOrder extends Algorithm{
 
   setObj(new_var){
     this.objective = new_var
-    this.x = this.x_ini.map(x => x);
     this.reinitialize();
   }
 
   setXini(new_var){
     this.x_ini = new_var.map(val => parseFloat(val));
-    this.x = this.x_ini.map(x => x);
     this.reinitialize();
   }
 
   setStep(new_var){
     this.delta = new_var;
-    this.x = this.x_ini.map(x => x);
     this.reinitialize();
   }
 
   reinitialize() {
+    this.x = this.x_ini.map(x => x);
     this.path = [];
   }
 }
 
+/** Mother class First Order algorithm must do the following task
+* @method hessian : Approximate the Hessian our implementation only works for 1 element and 2 element arrays
+*/
+class AlgorithmSecondOrder extends AlgorithmFirstOrder{
+  constructor(objective, x_ini, h = 0.001, delta = 0.1){
+    super(objective, x_ini, h = 0.001, delta = 0.1);
+  }
+
+  hessian(arr) {
+    let n = arr.length;
+    if (n == 1){
+      return [[(- 1 * this.objective([arr[0] + 2*this.h]) + 16 * this.objective([arr[0] + this.h]) - 30 * this.objective(arr) + 16 * this.objective([arr[0] - this.h]) - this.objective([arr[0] - 2*this.h]))/(12 * this.h ** 2)]];
+    } else if (n == 2){
```

Comment:
```suggestion
}
```
You don't need the "else if" here as the first if finishes by a return statement.
You can throw an error at the beginning of the function in case n > 2.